### ISSUE TITLE
pre_get_posts filter should not be flagged to return value

### ISSUE THREAD

**Issue by rebeccahum** [Automattic/VIP-Coding-Standards#394]
## Bug Description

```
Warning: Please, make sure that a callback to 'pre_get_posts' filter is always returning some value.
```

Since it's an action and the argument is passed by reference, return is not necessary.

## Minimal Code Snippet

```php
<?php 

add_filter( 'pre_get_posts', 'test_function' );

function test_function( $test ) {
	// Do something.
}
```

## Error Code

`WordPressVIPMinimum.Hooks.AlwaysReturnInFilter.MissingReturnStatement`

## Tested Against `master` branch?

- [x] I have verified the issue still exists in the `master` branch of VIPCS.

**Comment by GaryJones** [comment]
We looked into this and considered adding a configurable list of known action hooks (e.g. `pre_get_posts`, `parse_query`, `pre_get_users`) that the sniff would skip. However, maintaining such a list is inherently fragile — it would always be incomplete, and custom action hooks would need manual configuration. It shifts the burden rather than solving the problem.

The root cause here is that `add_filter()` is being used to register a callback for an action hook. Since the sniff only processes `add_filter()` calls, it reasonably expects a return value. The fix is to use `add_action()` instead, which is semantically correct and avoids the false positive entirely.

One thing we could do is update the error message to acknowledge both possibilities — a genuinely missing return in a filter callback, or an action hook incorrectly registered via `add_filter()`. Something along the lines of:

> The callback for the `%s` hook does not return a value. If this is a filter, callbacks must always return a value. If this is an action, use `add_action()` instead of `add_filter()`.

That said, we're going to close this out as working as intended for now. The sniff is correctly flagging code that uses `add_filter()` without returning a value, and the message already draws attention to the issue.
